Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/mtl_state_cache.mm

```diff
@@ -943,6 +943,60 @@ void ToObjC(const RenderPassStencilAttachmentDesc &desc,
     return re.first->second;
 }
 
+static bool ValidateRenderPipelineState(const MTLRenderPipelineDescriptor *descriptor,
+                                        ContextMtl *context,
+                                        const mtl::ContextDevice &device)
+{
+    // Ensure there is at least one valid render target.
+    bool hasValidRenderTarget = false;
+
+    for (NSUInteger i = 0;
+         !hasValidRenderTarget && i < GetMaxNumberOfRenderTargetsForDevice(device); ++i)
```

Comment:
I know the other places do it like this, but maybe you could set the precedent to query the count before the loop.

Reply:
I purposely searched how loop counts were written to keep it in the same style as the rest of ANGLE.